### DISCUSSION TITLE
rux-pop-up enhancements

### DISCUSSION THREAD

**Discussion by kiley-mitti** [RocketCommunicationsInc/astro#957] [Feature Requests]
Just dropping a note here for documentation purposes about a couple of enhancement to run-pop-up that might be looked into.

1) New Prop: activation-on (or just activation), default would activate the pop up on click as it currently does. another option could be hover or focus which would allow the popup to appear when one mouses over or focuses on the trigger.

2) Look into the way rux-pop-up interacts with agGrid. There are a couple of oddities in the way agGrid and rux-pop-up work together. For instance, if you want to place an agGrid table within a popup you must set a height for the pop up itself as agGrid sets itself to 100% and will not appear if an outer height is not set.

**Comment by micahjones13** [comment]
In regards to the first item, I feel like using the show and hide methods that we provide would be an easy way to accomplish opening on hover or some other way, no? Adding an `activation-on` and allowing for a mouse hover or something to open the pop-up is just a tool-tip, and we're building that already.  
Apologies if these have already been addressed since this was written so long ago

**Reply by kiley-mitti** [reply]
> In regards to the first item, I feel like using the show and hide methods that we provide would be an easy way to accomplish opening on hover or some other way, no?

That's a good point. It could be built out to do an on hover by the dev since they have access to these events. :)
